### PR TITLE
feat: replace create event button with LiquidButton

### LINKED ISSUE #158: Implement the composable button with glass Effect
Implement the composable button with a glass effect as shown in the Figma design. This composable will be reused across different screens.

Example in the Figma:
Image

### PR DESCRIPTION
**Pull request by poujhh**
Introduces the `LiquidButton`, a new composable with a liquid glass effect, to enhance the UI.

This change replaces the standard `Button` on the `MapScreen` used for event creation with the new `LiquidButton`. The `LiquidButton` utilizes the `AndroidLiquidGlass` library to create a visually appealing, interactive button with blur and lens effects. 

- The color is changeable by the caller 
- Unified opacity with the bottom navigation bar
- Created a COMMON_CAPSULE shape for the RoundedCornerShape specific to the navigation bar and the button for less code duplication.
- Moved the Map create event button to the scaffhold for better background coherence

image_2025-11-15_171433292

image_2025-11-15_171452782

closes #158

### CODE REVIEW

File: app/src/main/java/com/android/universe/ui/components/LiquidButton.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                Modifier
+              })
+          .height(48f.dp)
+          .padding(horizontal = 16f.dp),
+      horizontalArrangement = Arrangement.spacedBy(8f.dp, Alignment.CenterHorizontally),
+      verticalAlignment = Alignment.CenterVertically,
+      content = content)
+}
+
+object LiquidButtonTestTags {
```
Comment (Corentin00) [inline app/src/main/java/com/android/universe/ui/components/LiquidButton.kt]: The responsibility for the test tag should always lie with the caller, because imagine we have multiple buttons on the same screen: line 88 will force all buttons to have the same test tag. So we need to remove the `LiquidButtonTestTags` object and line 88, and let the caller handle this by passing a modifier with a test tag to its `LiquidButton` instance.

File: app/src/test/java/com/android/universe/ui/components/LiquidButtonTest.kt
Comment (Corentin00) [inline app/src/test/java/com/android/universe/ui/components/LiquidButtonTest.kt]: I would expect the test to be a little more comprehensive, for example by testing edge cases such as:

- Passing a very long text but asking a very small button to display it to see how it is displayed.
- Testing a click when the button is not enabled.
- Testing a click when `isInteractive` is `false`.

File: app/src/main/java/com/android/universe/ui/components/LiquidButton.kt [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+  val interactiveHighlight =
+      remember(animationScope) { InteractiveHighlight(animationScope = animationScope) }
+
+  Row(
+      modifier
+          .testTag(LiquidButtonTestTags.LIQUID_BUTTON)
+          .drawBackdrop(
+              backdrop = backdrop,
+              shape = { RoundedCornerShape(percent = 50) },
```
[the remark this answers is not in the document]
Comment (poujhh) [reply app/src/main/java/com/android/universe/ui/components/LiquidButton.kt]: You're right i've created a Shaped file where we can put our common shapes and put it in there for easy access

File: app/src/main/java/com/android/universe/ui/components/LiquidButton.kt [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+ *   background layer that the component's glass effects will interact with.
+ * @param modifier The [Modifier] to be applied to the component's container.
+ * @param isInteractive Whether the component is interactive or not.
+ * @param tint The color to tint the component with.
+ * @param surfaceColor The color to fill the component with.
+ * @param content The composable content for the button, typically a series of `Icon` or `Text`
+ *   composables. This lambda is executed within a [RowScope].
+ */
+@Composable
+fun LiquidButton(
```
[the remark this answers is not in the document]
Comment (poujhh) [reply app/src/main/java/com/android/universe/ui/components/LiquidButton.kt]: Indeed i've noticed that when trying to add a width modifier or height they are applied before everything and thus breaking the form. So i've thus made explicit parameters to the button now

File: app/src/test/java/com/android/universe/ui/components/LiquidButtonTest.kt
Comment (poujhh) [reply app/src/test/java/com/android/universe/ui/components/LiquidButtonTest.kt]: Indeed that made me think i forgot to put the enable parameter to the button. And for the very long text, already by default it clips the text so i don't see this as useful test

File: app/src/main/java/com/android/universe/ui/components/LiquidButton.kt [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+              role = Role.Button,
+              onClick = onClick)
+          .then(
+              if (isInteractive) {
+                Modifier.then(interactiveHighlight.modifier)
+                    .then(interactiveHighlight.gestureModifier)
+              } else {
+                Modifier
+              })
+          .height(48f.dp)
```
[the remark this answers is not in the document]
Comment (poujhh) [reply app/src/main/java/com/android/universe/ui/components/LiquidButton.kt]: indeed it's now defaulted and by parameter

File: app/src/main/java/com/android/universe/ui/map/MapScreen.kt
[the remark this answers is not in the document]
Comment (poujhh) [reply app/src/main/java/com/android/universe/ui/map/MapScreen.kt]: Actually it was an opcacity problem. it's now as readable as in the images. what do you think ?

**Review by Corentin00 — CHANGES_REQUESTED** [review]
Just rename something and everything will be fine.

**Review by Corentin00 — APPROVED** [review]
